virttest/utils_net.py

```diff
@@ -3193,34 +3200,40 @@ def gen_ipv4_addr(network_num="10.0.0.0", network_prefix="24", exclude_ips=[]):
             yield str(ip_address)
 
 
-def get_ip_address_by_interface(ifname, ip_ver="ipv4"):
+def get_ip_address_by_interface(ifname, ip_ver="ipv4", linklocal=False):
     """
     returns ip address by interface
     :param ifname: interface name
     :param ip_ver: Host IP version, ipv4 or ipv6.
+    :param linklocal: Whether ip address is local or remote
     :raise NetError: When failed to fetch IP address (ioctl raised IOError.).
     """
     if ip_ver == "ipv6":
         ver = netifaces.AF_INET6
     else:
         ver = netifaces.AF_INET
     addr = netifaces.ifaddresses(ifname).get(ver)
+
     if addr is not None:
-        return [a['addr'] for a in addr if not a['addr'].startswith('fe80')][0]
+        if linklocal:
+            return [a['addr'] for a in addr if a['addr'].startswith('fe80')][0]
```

Comment:
Well ipv4 has link local addr in other form, so need to distinguish it with ipv6's.

see https://tools.ietf.org/html/rfc3927